documents/int256.go

```diff
@@ -0,0 +1,108 @@
+package documents
+
+import (
+	"math/big"
+	"regexp"
+
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/utils/byteutils"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// signByteIdx is the index of byte that contains the sign bit of the int256 when represented as 32 bytes
+const signByteIdx = 0
+
+// validInt regex for only allowing integers
+var validInt = regexp.MustCompile(`^[-+]?\d+$`)
+
+// Int256 represents a signed 256 bit integer
+type Int256 struct {
+	v big.Int
+}
+
+// NewInt256 creates a new Int256 given a string
+func NewInt256(n string) (*Int256, error) {
+	if !validInt.MatchString(n) {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("probably a decimal value: %s", n))
+	}
+
+	nn := new(big.Int)
+	nn, ok := nn.SetString(n, 10)
+	if !ok {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("probably an arbitrary string: %s", n))
+	}
+
+	if !isValidInt256(*nn) {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %s", n))
+	}
+	return &Int256{*nn}, nil
+}
+
+// FromBytes converts the a big endian byte slice to an Int256
+func FromBytes(b []byte) (*Int256, error) {
+	if len(b) != 32 {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %x", b))
+	}
+
+	// check and record the sign bit
+	neg := false
+	if byteutils.IsBitSet(b[signByteIdx], 7) {
+		neg = true
+	}
+	// ignore the sign bit
+	b[signByteIdx] = byteutils.ClearBit(b[signByteIdx], 7)
+
+	nn := new(big.Int)
+	nn.SetBytes(b[:])
+	if !isValidInt256(*nn) {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %s", nn.String()))
+	}
+
+	// negative
+	if neg {
+		nn = nn.Neg(nn)
+	}
+
+	return &Int256{*nn}, nil
+}
+
+// Bytes returns the big endian 32 byte representation of this int256. First bit of LSB is used as the sign bit.
+func (i *Int256) Bytes() [32]byte {
+
+	var b [32]byte
+	// no of bits in i.v.Bytes() <= 255
+	// if its less pad the number in big endian order and copy to the 32 byte array
+	copy(b[:], math.PaddedBigBytes(&i.v, 32))
+
+	//  set the sign bit
+	b[signByteIdx] = byteutils.ClearBit(b[signByteIdx], 7)
```

Comment:
If the opertions on Int256 leads to result above max value, this function should error out instead of ignoring the excess bytes. Else, we might end up with unexpected results

Reply:
As the comment on line 73 says it's an invariant for int256 type that is enforced, this function shouldn't care about the size. When we have more ops we can add the check to those as well as I do now for `NewInt256`.

Reply:
Sure, lets add a todo here so that we track this. Better yet, change the function signature to return an error so that we dont have to refactor everything when that check comes in.

But this is check we must do though

Reply:
As said the check is not related to this function, it will be on those `op` functions. Right now its impossible for you to create a value more than 256 bits, unless someone hacks the internal big int which we must not do.